css/src/components/modal.css

```diff
@@ -56,13 +56,37 @@
 }
 
 .Modal-header {
-  padding: var(--spacing-2) var(--spacing-2) var(--spacing-l) var(--spacing-xl);
+  padding: var(--spacing-2) var(--spacing-2) var(--spacing-l) 0;
+  box-sizing: border-box;
+}
+
+.Modal-header--seperator {
+  border-bottom: var(--border);
+}
+
+.Modal-headerWrapper {
+  display: flex;
+  align-items: center;
+  margin-left: var(--spacing-xl);
+}
+
+.Modal-headerWrapper--backIcon {
+  margin-left: var(--spacing-2);
 }
 
 .Modal-headerHeading {
   display: flex;
   align-items: center;
   justify-content: space-between;
+  width: 100%;
+}
+
+.Modal-headerBackIcon {
+  height: var(--spacing-3);
+  width: var(--spacing-3);
+  display: flex;
+  align-items: center;
+  justify-content: center;
 }
```

Comment:
Pass size and different utility classes to `Icon` component only.

Reply:
Icon has a margin of `6px` which is not used in css tokens. Hence it is better to use a icon wrapper instead of adding different utility classes to Icon